server/async-worker/index.js

```diff
@@ -0,0 +1,65 @@
+const { Op } = require("sequelize");
+const callScraper = require("../scraper");
+const { Mention, User } = require("../models");
+
+// A rough draft of how new mentions would get pulled into the db
+// without using an explicit /POST route as well as using
+// a given user's credentials.
+module.exports = async function asyncWorker() {
+  const SECONDS = 1;
+  const MILLISECONDS = 1000;
+  // TODO refactor this into smaller functions
+  console.log("[Scraper] Calling async scraper");
+
+  const users = await User.findAll();
+  for (let user of users) {
+    let mentions = await callScraper(user.company);
+    let count = 0;
+
+    for (let m of mentions) {
+      let col, val, timeConvention;
+      // if this is a twitter mention
+      if (m.platform == "Twitter") {
+        col = "content";
+        val = m.content;
+        timeConvention = SECONDS;
+      }
+      // if this is a reddit mention
+      else if (m.platform == "Reddit") {
+        col = "title";
+        val = m.title;
+        timeConvention = MILLISECONDS;
+      }
+
+      let userMentions = await user.getMentions({
+        where: {
+          [col]: val,
+        },
+      });
```

Comment:
I'm going to be removing this check, because as @powerseed this turned out to be extraneous as well. As it turns out, the method called below, `user.addMention(mention)` will not add duplicates of the mention already exists for the user so this block is not necessary. I believe the initial reason I had it in the first place was because I was not aware that the method added by sequelize took care of it